source/core_json.c

```diff
@@ -1138,6 +1167,55 @@ JSONStatus_t JSON_Validate( const char * buf,
 
 /** @cond DO_NOT_DOCUMENT */
 
+/**
+ * @brief Output index and length for the next value.
+ *
+ * Also advances the buffer index beyond the value.
+ * The value may be a scalar or a collection.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ * @param[out] value  A pointer to receive the index of the value.
+ * @param[out] valueLength  A pointer to receive the length of the value.
+ *
+ * @return true if a value was present;
+ * false otherwise.
+ */
+static bool_ nextValue( const char * buf,
+                        size_t * start,
```

Comment:
Better naming suggestion
```suggestion
                        size_t * startIndex,
```

Reply:
I think `start` is fine.  It's used in the majority of functions in the same fashion.